rust/scx_rustland_core/assets/bpf/main.bpf.c

```diff
@@ -479,18 +485,41 @@ s32 BPF_STRUCT_OPS(rustland_select_cpu, struct task_struct *p, s32 prev_cpu,
 	if (full_user)
 		return prev_cpu;
 
+	tctx = lookup_task_ctx(p);
+	if (!tctx)
+		return prev_cpu;
+
+	/*
+	 * If the previously used CPU is still available, keep using it to take
+	 * advantage of the cached working set.
+	 */
+	if (scx_bpf_test_and_clear_cpu_idle(prev_cpu)) {
+		tctx->dispatch_local = true;
```

Comment:
BTW, you can now directly call `scx_bpf_dispatch()` from `select_cpu()`, so you don' t have to set `dispatch_local` to tell `enqueue()`.

Reply:
Yeah, I can definitely keep the flag to track the direct dispatches, and then dispatch directly from `select_cpu()`... I guess my brain is just too used to the "old" dispatch_local way :)